### PR TITLE
Travis build now fails if coverage is less than 100%

### PR DESCRIPTION
**Pull request by wizmer**
The coverage command was performed in stage 'after_success' but its return
code is ignored so it was basically doing nothing.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,11 +18,9 @@ install:
   - pip install 'matplotlib==1.5.2' tox
   - pip install .
 script:
-  - if [[ $TRAVIS_PYTHON_VERSION == 2.7 ]]; then tox -e py27; fi
+  - if [[ $TRAVIS_PYTHON_VERSION == 2.7 ]]; then tox -e py27 -e coverage; fi
   - if [[ $TRAVIS_PYTHON_VERSION == 3.5 ]]; then tox -e py35; fi
   - if [[ $TRAVIS_PYTHON_VERSION == 3.6 ]]; then tox -e py36 -e lint; fi
```
Comment (mgeplf) [inline .travis.yml]: should we be doing the coverage on a newer python?

Comment (wizmer) [reply .travis.yml]: Good point. I changed that.

**Comment by wizmer** [comment]
I also changed also the default tox env to python3, hence the changes in the source files.

**Comment by mgeplf** [comment]
Yup, great idea.  We could probably add newer python versions too; but we can do that in another PR, if you want.

**Comment by wizmer** [comment]
I'll take the opportunity of the failing Travis build to try some new stuff. It may take a while. You can unsubscribe from this PR if you want. I'll remove you from the reviewers.

File: .travis.yml [hunk 2]
```diff
@@ -1,28 +1,13 @@
...
-    - gfortran
-    - libmpfr-dev
-    - libhdf5-dev
+  - "2.7"
+  - "3.6"
+  - "3.7"
 cache: pip
 install:
-  - pip install 'matplotlib==1.5.2' tox
-  - pip install .
```
Comment (wizmer) [inline .travis.yml]: I'm wondering if this line is required for the deployement stage ?

**Comment by wizmer** [comment]
@mgeplf That should be good now.
I have added thet use of tox-travis to manage the Travis tox envs.
Also I replaced the py35 by py37.

File: tox.ini [hunk 3]
```diff
@@ -7,43 +7,37 @@ testdeps =
...
 [tox]
 envlist =
-    lint
-    py27
-    py36
-    coverage
+    {py27,py36}
+    py36-lint
+    py36-coverage
+    py36-docs
```
Comment (wizmer) [inline tox.ini]: Does the 'docs' env name matters for readthedoc ?

Comment (haleepfl) [reply tox.ini]: readthedocs doesn't use tox, so it shouldn't be a problem

File: .travis.yml (hunk 2 above)
Comment (mgeplf) [reply .travis.yml]: iirc, pinning the matplotlib version was to make sure the tests passed

**Pull request merged**